### PR TITLE
fix(accounts-db): misc fixes

### PR DESCRIPTION
**Pull request by 0xNineteen**
- move diskAllocator from index.zig into allocators.zig 
- fix disk reference allocator 
- fix file_map.deinit with incorrect allocator 
- add additional test to RecycleFBA
- add geyser support to `AccountsDB.putAccountSlice`
- add cli flag for `accounts_per_file_estimate` (used for preallocation during indexing) 
- move many root folders into `misc/` dir
- change all folders to use snake_case
- add global constants for paths in sig.zig (eg, `pub const TEST_DATA_DIR = "misc/test_data/";`)
- change main validator output directory from `ledger/` to `validator/` (ledger is used elsewhere in blockstore so doesnt make sense to have it as the root dir for all validator related files)

### CODE REVIEW

**Review by InKryption — CHANGES_REQUESTED** [review]
Just one observation that seems like it would be good to include in this PR. I like the centralization of the "canonical directories" we use.

**Review by yewman — DISMISSED** [review]
lgtm

File: src/utils/allocators.zig [hunk 1]
```diff
@@ -117,3 +118,215 @@ pub const RecycleFBA = struct {
...
+    pub fn init(filepath: []const u8) Self {
+        return Self{
+            .filepath = filepath,
+        };
+    }
+
+    /// deletes all allocated files + optionally frees the filepath with the allocator
+    pub fn deinit(self: *Self, str_allocator: ?std.mem.Allocator) void {
+        self.mux.lock();
+        defer self.mux.unlock();
```
[the remark this answers is not in the document]
Comment (0xNineteen) [reply src/utils/allocators.zig]: f51c761

**Review by InKryption — APPROVED** [review]
LGTM

**Pull request merged**